lutris/gui/views/base.py

```diff
@@ -28,35 +28,57 @@ def popup_contextual_menu(self, view, event):
         """Contextual menu."""
         if event.button != Gdk.BUTTON_SECONDARY:
             return
-        view.current_path = view.get_path_at_pos(event.x, event.y)
-        if view.current_path:
-            view.select()
-            model = self.get_model()
-            _iter = model.get_iter(view.current_path[0])
-            if not _iter:
-                return
-            col_id = str(model.get_value(_iter, COL_ID))
-
-            game_actions = self.get_game_actions(col_id)
+        game_ids = []
+        if self.view_type == "grid":
```

Comment:
If you want to check the type of the view, you can use isinstance(self, GameGridView); no need to track object types with a field like this. But it would be better to use methods on the view classes- you can have matching methods on both view classes, that provide the selection in whatever way is required. Polymorphism!

Reply:
Thank you. Will keep that in mind

Reply:
Tried to use isinstace(self, GameGridView). GameGridView is undefined in GameView it seems.

Reply:
Hmm. I see the problem. You can't import GameGridView because that forms a cycle, a circular pattern of imports, and Python does not allow that. I'm used to languages that do.

There are several ways to fix that. The correct way is to move to move the code to the data. Let's say: create a get_selected_game_actions() method on both GameGridView and GameListView, then call it via 'self.get_selected_game_actions()' in GridView. That is a classic refactoring.

This method would do all the selection jiggering and would return the suitable GameActions object.

However, if you find that too challenging to tackle, leave it the way it is and I will sort this out after the merge.

(I'll probably want to make more commands work in the multi-select case, like 'Add to Favories', too. But lets get 'Remove' done first.)